src/js/controller/instream-adapter.js

```diff
@@ -98,6 +99,12 @@ const InstreamAdapter = function(_controller, _model, _view, _mediaPool) {
         // Make sure the original player's provider stops broadcasting events (pseudo-lock...)
         _controller.detachMedia();
 
+        // Shaka cleans the videotag async, so wait until its destroy promise fulfills.
+        const provider = _model.get('provider');
+        if (!_model.get('backgroundLoading') && provider && provider.name === 'shaka') {
+            _destroyPromise = _model._provider.destroy();
```

Comment:
This would result in a second call to provider.destroy(). Can we get the promise from the first one (if `detachMedia` returns a promise, then we have a destroy promise)?

Reply:
`detachMedia` assigns `programController.attached = false`, which is a setter. This setter calls `provider.destroy`, so I'm not sure how we could get the destroy promise this way.

Reply:
Change it to a method (+ read-only getter). We should not use setters for async actions.